frontend/src/views/WheresMyGene/components/GeneSearchBar/index.tsx

```diff
@@ -160,35 +211,124 @@ export default function GeneSearchBar({ onGenesChange }: Props): JSX.Element {
     onGenesChange(selectedGenes);
   }, [onGenesChange, selectedGenes]);
 
-  useEffect(() => {
-    if (Number.isNaN(Number(input))) return;
-
-    setSelectedGenes(genes.slice(0, Number(input)));
-  }, [input, genes]);
+  const handleClose = (_, reason: AutocompleteCloseReason) => {
+    if (reason === "toggleInput") {
+      return;
+    }
+    setOpen(false);
+  };
+  const handleChange = (_, newValue: DefaultMenuSelectOption[] | null) => {
+    return setSelectedGenes(newValue as Gene[]);
+  };
+  const handleClick = () => {
+    setOpen(true);
+  };
+
+  const handlePaste = () => {
+    setPendingPaste(true);
+  };
+
+  const handleEnter = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter" && pendingPaste) {
+      event.preventDefault();
+      const newSelectedGenes = [...selectedGenes];
+      const pastedGenes = pull(uniq(input.split(/[ ,]+/)), "");
+      const allGenes = genes.reduce((acc, gene) => {
+        return acc.set(gene.name, gene);
+      }, new Map<Gene["name"], Gene>());
+      pastedGenes.map((gene) => {
+        const newGene = allGenes.get(gene);
+        if (!newGene) {
+          Toast.show({
+            intent: Intent.DANGER,
+            message: `Gene not found: ${gene}`,
+          });
+        } else if (!newSelectedGenes.includes(newGene))
+          newSelectedGenes.push(newGene);
+      });
+      setPendingPaste(false);
+      setOpen(false);
+      return setSelectedGenes(newSelectedGenes);
+    }
+  };
+
+  const handleInput = (
+    _,
```

Comment:
```
[2] src/views/WheresMyGene/components/GeneSearchBar/index.tsx(315,11): error TS2322: Type '(_: any, value: string, reason: AutocompleteInputChangeReason) => void' is not assignable to type '((event: ChangeEvent<{}>, value: string, reason: AutocompleteInputChangeReason) => void) & ((event: SyntheticEvent<Element, Event>) => void)'.
[2]   Type '(_: any, value: string, reason: AutocompleteInputChangeReason) => void' is not assignable to type '(event: SyntheticEvent<Element, Event>) => void'.
```

Can we suppress type checks on vars named `_`?

Reply:
Interesting, we should have lint rule to suppress it now 🤔 

```ts
"@typescript-eslint/no-unused-vars": [
      "error",
      {
        args: "after-used",
        argsIgnorePattern: "^_",
        ignoreRestSiblings: false,
        vars: "all",
      },
    ],
```

Reply:
Wouldn't that just suppress no-unused-vars and not error `TS2322`

Reply:
ooh I thought `no-unused-vars` would suppress type check. Guess not 😆  I didn't find a way to do that, so maybe just add the event type for now and move on?

Reply:
`ChangeEvent<{}>` also results in an error due to the empty object 🙃

Reply:
Ah darn it lol

Actually, since we're merging https://github.com/chanzuckerberg/sci-components/pull/122, maybe we can just use `InputBaseProps` instead of `onInputChange`, since its typing is incorrect?

```ts
const InputBaseProps = useMemo(() => {
  return {
    placeholder: "...",
    onChange: handleInput,
  }
}, [...])
<MenuSelect
  InputBaseProps={}
/>
```

`onInputChange()`'s type needs to be changed from `(event: React.SyntheticEvent) => void;`  to `React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>` I believe. But not sure if that will break Aspen's code, so we can do that in SDS later!

Reply:
Sounds good! What does the series of actions look like in the case? I wait for merge, then a release is cut for SDS, and then we update our import and I alter this PR's branch?

Reply:
That's right! v1.12.0 has been automatically released with the [merge](https://github.com/chanzuckerberg/sci-components/pull/122#issuecomment-1048110374), so we should be able to update SDS in DP and use it now 🎉

Reply:
This is actually incorrect! Autocorrect's `onChange` prop lives on the parent component, because it is an extension of `InputBaseProps'` `onChange`(addition of value and reason args)

Reply:
I put some more time into trying to type this and still can't get this to work. @tihuan Could you give this a try?

Reply:
based on Bruce's comment on the sds slack channel, maybe this is typing error on sds? https://github.com/chanzuckerberg/sci-components/blob/main/src/core/MenuSelect/index.tsx